### PR TITLE
bcachefs: add support for subvolumes

### PR DESCRIPTION
**Pull request by RaitoBezarius**
Relies on unreleased NixOS commits.

### CODE REVIEW

File: lib/types/table.nix [hunk 1]
```diff
@@ -25,7 +25,7 @@
             description = "Partition type";
           };
           fs-type = lib.mkOption {
-            type = lib.types.nullOr (lib.types.enum [ "btrfs" "ext2" "ext3" "ext4" "fat16" "fat32" "hfs" "hfs+" "linux-swap" "ntfs" "reiserfs" "udf" "xfs" ]);
+            type = lib.types.nullOr (lib.types.enum [ "bcachefs" "btrfs" "ext2" "ext3" "ext4" "fat16" "fat32" "hfs" "hfs+" "linux-swap" "ntfs" "reiserfs" "udf" "xfs" ]);
```
Comment (Lassulus) [inline lib/types/table.nix:28]: thats not needed, it's only for dos partition tables

Comment (RaitoBezarius) [reply lib/types/table.nix:28]: gotcha

File: lib/types/bcachefs.nix [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+          };
+          extraArgs = lib.mkOption {
+            type = lib.types.listOf lib.types.str;
+            default = [ ];
+            description = "Extra arguments";
+          };
+          mountpoint = lib.mkOption {
+            type = lib.types.nullOr diskoLib.optionTypes.absolute-pathname;
+            default = null;
+            description = "Location to mount the subvolume to.";
```
Comment (Lassulus) [inline lib/types/bcachefs.nix:46]: if mountpoint has no effect, why not remove it?

Comment (RaitoBezarius) [reply lib/types/bcachefs.nix:46]: yep, I think it should just be the name

**Comment by Mic92** [comment]
Continues in: https://github.com/nix-community/disko/pull/499

**Pull request closed without merging**